hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/translators/BaseTranslator.java

```diff
@@ -153,8 +179,39 @@ public void prepareForUnit(@NonNull final BlockTransactionalUnit unit) {
                 nextCreatedNums.values().stream().mapToLong(List::getLast).max().orElse(highestKnownEntityNum);
     }
 
+    /**
+     * Determines if the given number was created in the ongoing transactional unit.
+     *
+     * @param num the number to query
+     * @return true if the number was created
+     */
+    public boolean createdThisUnit(final long num) {
+        return num > prevHighestKnownEntityNum;
+    }
+
+    /**
+     * Tracks the association of a token with an account.
+     *
+     * @param tokenID the token to track
+     * @param accountID the account to track
+     */
+    public void trackAssociation(@NonNull final TokenID tokenID, @NonNull final AccountID accountID) {
+        knownAssociations.add(new TokenAssociation(tokenID, accountID));
+    }
+
+    /**
+     * Tracks the dissociation of a token from an account.
+     *
+     * @param tokenID the token to track
+     * @param accountID the account to track
+     */
+    public void trackDissociation(@NonNull final TokenID tokenID, @NonNull final AccountID accountID) {
```

Comment:
Is this necessary? Doesn't seem to be utilized in `TokenDissociateTranslator `and `TransactionRecord `doesn't seem to have a field for disassociations.

Reply:
It's used [here](https://github.com/hashgraph/hedera-services/blob/15285-blk-rec-translator/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/translators/impl/TokenDissociateTranslator.java#L42); we need to keep up-to-date on what associations exist so we can classify automatic token associations correctly.